packages/flutter_tools/lib/src/version.dart

```diff
@@ -236,21 +236,37 @@ class FlutterVersion {
     final Duration frameworkAge = _clock.now().difference(localFrameworkCommitDate);
     final bool installationSeemsOutdated = frameworkAge > kVersionAgeConsideredUpToDate;
 
+    /// Gets whether or not there is a new version of Flutter available on the remote.
+    ///
+    /// Returns null if the cached version is out-of-date or missing, and we are
```

Comment:
If this can return tri-state, let's use an enum rather than a nullable boolean.

Reply:
Done; reads much nicer now 👍